architecture/2024-11-ucp-component-threat-model.md

```diff
@@ -0,0 +1,247 @@
+
+3. **Deployment Engine**: Deployment Engine makes requests to the UCP to deploy and also ask about the deployment status of certain objects within the ongoing deployment. Example:
+
+   ```text
+   {
+   "timestamp": "2024-11-03T14:06:54.684Z",
+   "severity": "INFO",
+   "message": "Start processing HTTP request GET https://ucp.radius-system/apis/api.ucp.dev/v1alpha3/planes/radius/local/resourceGroups/default/providers/Applications.Core/containers/demo?api-version=2023-10-01-preview",
+   "name": "RequestPipelineStart",
+   "scope": "System.Net.Http.HttpClient.local.LogicalHandler",
+   "serviceName": "deployment.engine",
+   "version": "0.38\u002B1b2449770894506793f865dc7ecc9ed8e6df0d93",
+   "hostName": "bicep-de-6b86cc859f-tt2r8",
+   "traceId": "a0d750fc4e5d35bd1c615d87c1261f72",
+   "spanId": "5b624ecddd0b19c2",
+   "HttpMethod": "GET",
+   "Uri": "https://ucp.radius-system/apis/api.ucp.dev/v1alpha3/planes/radius/local/resourceGroups/default/providers/Applications.Core/containers/demo?api-version=2023-10-01-preview"
+   },
+   {
+   "timestamp": "2024-11-03T14:06:54.684Z",
+   "severity": "INFO",
+   "message": "Sending HTTP request GET https://ucp.radius-system/apis/api.ucp.dev/v1alpha3/planes/radius/local/resourceGroups/default/providers/Applications.Core/containers/demo?api-version=2023-10-01-preview",
+   "name": "RequestStart",
+   "scope": "System.Net.Http.HttpClient.local.ClientHandler",
+   "serviceName": "deployment.engine",
+   "version": "0.38\u002B1b2449770894506793f865dc7ecc9ed8e6df0d93",
+   "hostName": "bicep-de-6b86cc859f-tt2r8",
+   "traceId": "a0d750fc4e5d35bd1c615d87c1261f72",
+   "spanId": "5b624ecddd0b19c2",
+   "HttpMethod": "GET",
+   "Uri": "https://ucp.radius-system/apis/api.ucp.dev/v1alpha3/planes/radius/local/resourceGroups/default/providers/Applications.Core/containers/demo?api-version=2023-10-01-preview"
+   }
+   ```
+
+4. **Health Check Probes**: Kubernetes itself can act as a client by performing health and readiness checks on the Universal Control Plane.
+
+5. **Metrics Scrapers**: If metrics are enabled, Prometheus or other monitoring tools can scrape metrics from the Universal Control Plane.
+
+## Trust Boundaries
+
+We have a few different trust boundaries for the UCP component:
+
+- **Kubernetes Cluster**: The overall environment where the UCP component operates and receives requests from the clients.
+- **Namespaces within the Cluster**: Logical partitions within the cluster to separate and isolate resources and workloads.
+
+The UCP component lives inside the `radius-system` namespace in the Kubernetes cluster where it is installed. Other components of Radius like Controller, Deployment Engine, Dashboard, and Applications RP also live in the same namespace.
+
```

Comment:
What can we say about the trust model of clients?

What about the trust model of resource providers?

What about the trust model of external resource managers? (Azure or AWS)

Reply:
Let's talk about the trust models of clients, resource providers, and external resource managers.

For resource providers, I think, the trust model is the `radius-system` namespace.

Some clients are in the same namespace, but some clients are not.

Reply:
Find out how auth works and how clients authenticate with UCP

Reply:
What do we believe to be true when we send requests to App-RP or AWS or Azure?

Reply:
Updated this part accordingly.